packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/toolbarModule.tests.js

```diff
@@ -1340,6 +1441,146 @@ testModule('Toolbar dialogs', dialogModuleConfig, () => {
     });
 });
 
+testModule('Toolbar AI dialog', dialogAiModuleConfig, () => {
+    QUnit.test('renders base controls in AI dialog', function(assert) {
+        this.options.items = [{ name: 'ai', commands: ['summarize'] }];
+        new Toolbar(this.quillMock, this.options);
+
+        const showSpy = sinon.spy(this.options.editorInstance, 'showAiDialog');
+
+        this.$element
+            .find(`.${TOOLBAR_FORMAT_WIDGET_CLASS} .${MENU_ITEM_CLASS}`)
+            .trigger('dxclick');
+
+        const $menuItem = $(`.${MENU_ITEM_CLASS}`).last();
+        $menuItem.trigger('dxclick');
+
+        assert.ok(showSpy.calledOnce, 'showAiDialog called');
+        assert.deepEqual(showSpy.firstCall.args[0], {
+            currentCommand: 'summarize',
+            currentCommandOption: undefined,
+            text: 'Test',
+            commandsMap: { summarize: {
+                name: 'summarize',
+                options: undefined,
+                text: 'Summarize'
+            } },
+        }, 'Correct config passed to dialog');
+    });
+
+    QUnit.test('renders AI menu with default commands & options when AI item is passed as string', function(assert) {
+        this.options.items = ['ai'];
+
+        new Toolbar(this.quillMock, this.options);
+
+        const $menu = this.$element.find(`.${MENU_CLASS}`);
+        assert.ok($menu.length, 'menu for "ai" was rendered');
+
+        const menuInstance = $menu.dxMenu('instance');
+        const dataSource = menuInstance.option('dataSource');
+        assert.ok(dataSource, 'has dataSource');
+
+        const rootItem = dataSource[0];
+        const commands = rootItem.items || [];
+
+        const actualCommands = commands.map(command => command.id);
+        assert.deepEqual(actualCommands, defaultAiCommands, 'commands match default list');
+
+        const changeStyleCommand = commands.find(command => command.id === 'changeStyle');
+        const changeStyleOptions = changeStyleCommand.items.map(option => option.id);
+        assert.deepEqual(changeStyleOptions, defaultAiChangeStyleOptions, 'changeStyle options match defaults');
+
+        const changeToneCommand = commands.find(command => command.id === 'changeTone');
+        const changeToneOptions = changeToneCommand.items.map(option => option.id);
+        assert.deepEqual(changeToneOptions, defaultAiChangeToneOptions, 'changeTone options match defaults');
+
+        const translateCommand = commands.find(command => command.id === 'translate');
+        const translateOptions = translateCommand.items.map(option => option.id);
+
+        assert.deepEqual(translateOptions, defaultAiTranslateOptions, 'translate options match defaults');
+    });
+
+    QUnit.test('renders AI menu with default commands & options when AI item is passed as object', function(assert) {
+        this.options.items = [{ name: 'ai' }];
```

Comment:
This test is a complete copypaste from the previous one besides this row. Let's use forEach